controllers/datadogmonitor/controller.go

```diff
@@ -224,36 +242,47 @@ func (r *Reconciler) update(logger logr.Logger, datadogMonitor *datadoghqv1alpha
 	}
 
 	// Update monitor in Datadog
-	if _, err := updateMonitor(r.datadogAuth, logger, r.datadogClient, datadogMonitor); err != nil {
+	m, err := updateMonitor(r.datadogAuth, logger, r.datadogClient, datadogMonitor)
+	if err != nil {
 		status.SyncStatus = datadoghqv1alpha1.SyncStatusUpdateError
 		return err
 	}
 
 	event := buildEventInfo(datadogMonitor.Name, datadogMonitor.Namespace, datadog.UpdateEvent)
 	r.recordEvent(datadogMonitor, event)
 
-	// Set Updated Condition
+	// Set updated condition
 	condition.UpdateDatadogMonitorConditions(status, now, datadoghqv1alpha1.DatadogMonitorConditionTypeUpdated, corev1.ConditionTrue, "DatadogMonitor Updated")
 	status.SyncStatus = datadoghqv1alpha1.SyncStatusOK
+	modifiedTime := metav1.NewTime(m.GetModified())
+	status.Modified = &modifiedTime
 	logger.Info("Updated DatadogMonitor", "Monitor Namespace", datadogMonitor.Namespace, "Monitor Name", datadogMonitor.Name, "Monitor ID", datadogMonitor.Status.ID)
 
 	return nil
 }
 
-func (r *Reconciler) get(logger logr.Logger, datadogMonitor *datadoghqv1alpha1.DatadogMonitor, status *datadoghqv1alpha1.DatadogMonitorStatus, now metav1.Time) error {
+func (r *Reconciler) get(logger logr.Logger, datadogMonitor *datadoghqv1alpha1.DatadogMonitor, status *datadoghqv1alpha1.DatadogMonitorStatus, now metav1.Time) (datadogapiclientv1.Monitor, error) {
 	// Get monitor from Datadog and update resource status if needed
 	m, err := getMonitor(r.datadogAuth, r.datadogClient, datadogMonitor.Status.ID)
 	if err != nil {
 		status.SyncStatus = datadoghqv1alpha1.SyncStatusGetError
-		return err
+		return m, err
 	}
+	return m, nil
+}
 
+// Helper functions to modify an API monitor object
+func checkMonitorDefSync(m datadogapiclientv1.Monitor, status *datadoghqv1alpha1.DatadogMonitorStatus) bool {
+	monitorModifiedTime := m.GetModified()
```

Comment:
Why not compare monitor parameters (just 6, [here](https://github.com/DataDog/datadog-operator/blob/19b5666dbcbf3dfc1add46f2306b9da48a73da95/controllers/datadogmonitor/monitor.go#L24-L29)) for equality instead of relying on modification time (and window)?